packages/eslint-plugin-template/src/rules/i18n.ts

```diff
@@ -306,12 +324,14 @@ export default createESLintRule<Options, MessageIds>({
 
     return {
       ...(checkId && {
-        'Element[i18n]'(node: StronglyTypedElement) {
-          handleElement(node);
+        ':matches(Element, Template[tagName="ng-template"])[i18n]'(
+          node: StronglyTypedElement,
+        ) {
+          handleElementOrTemplate(node);
         },
       }),
       ...((checkAttributes || checkId) && {
-        [`Element > TextAttribute[value=${PL_PATTERN}]`](
+        [`:matches(Element, Template[tagName="ng-template"]) > TextAttribute[value=${PL_PATTERN}]`](
```

Comment:
Interestingly the compiler may return `Template` with a different tag name than `ng-template`, when using, for example, an element with structural directive, ex:

```html
<!-- The return here is `Template[tagName="mat-option"]` -->
<mat-option *ngFor="let item of items">{{ item.value }}</mat-option>
```